src/NuGet.Clients/NuGet.PackageManagement.UI/Actions/UIActionEngine.cs

```diff
@@ -1065,6 +1079,18 @@ private PackageLicenseInfo GeneratePackageLicenseInfo(IPackageSearchMetadata met
             return results;
         }
 
+        private static async Task<IPackageSearchMetadata[]> GetPackageMetadataThrottledAsync(IEnumerable<SourceRepository> sources, SourceCacheContext sourceCacheContext, IEnumerable<PackageIdentity> packages, CancellationToken token)
+        {
+#pragma warning disable CS8619 // Nullability of reference types in value doesn't match target type.
+            IPackageSearchMetadata[] remoteResults = (await TaskCombinators.ThrottledAsync(
+                packages,
+                (p, t) => GetPackageMetadataAsync(sources, sourceCacheContext, p, t),
+                token)).Where(metadata => metadata != null).ToArray();
+#pragma warning restore CS8619 // Nullability of reference types in value doesn't match target type.
```

Comment:
This code is just me extracting the existing code to a method. I didn't change anything about nullability. If you think I did, I'll take a look next week.

Reply:
I think it's a good idea to try and fix this code debt, if it's too complex maybe we can create an issue, so we are aware of it.

Reply:
What code debt are you referring to?

Reply:
For me when I see a "warning disable" is a sign that something we are doing is not good and at that time we decided that the best was just ignore it, probably this a needed disable to which I prefer using the file ones rather than source like this one.

And about the fix, I had the impression that we agreed that if we saw a code debt we were going to try and fix it if it was easy or create an issue (that probably will end in P3 icebox 😢 )

Reply:
```suggestion
            IPackageSearchMetadata[] remoteResults = (await TaskCombinators.ThrottledAsync(
                packages,
                (p, t) => GetPackageMetadataAsync(sources, sourceCacheContext, p, t),
                token)).Where(metadata => metadata != null).Cast<IPackageSearchMetadata>().ToArray();
```

I had a quick look in the IDE. `GetPackageMetadataAsync` returns `IPackageSearchMetadata?`, so the null checker is warning about `IPackageSearchMetadata?[]` being assigned to `IPackageSearchMetadata[]`. Using `.Cast<IPackageSearchMetadata>()` we can remove the "is allowed to be null" annotation.

Another option that might work (I didn't test) is to add our own LINQ extension `IEnumerable<T> WhereNotNull<T>(this IEnumerable<T?> incoming) { foreach (T? t in incoming) { if (t != default) { yield return t; } } }`

Basically, the way the current code is written, the compiler can't figure out nullability. We have to learn to write code a little differently to let the compiler check for us.